sql/core/src/test/scala/org/apache/spark/sql/connector/MergeIntoDataFrameSuite.scala

```diff
@@ -959,7 +960,9 @@ class MergeIntoDataFrameSuite extends RowLevelOperationSuiteBase {
         .insertAll()
         .whenNotMatchedBySource(col("col") === 1)
         .delete()
+        .asInstanceOf[MergeIntoWriterImpl[Row]]
       val writer2 = writer1.withSchemaEvolution()
```

Comment:
Now the implementation of `withSchemaEvolution()` would not duplicate the writer instance anymore. Would this code turn on schema evolution for `writer1` too?

Reply:
It is weird to me that everything else in the builder is mutable besides the schema evolution flag. It should be either fully immutable, or not at all... I lean towards mutability because all of the other builders are.

Reply:
Can you add an assertion to make sure writer1.schemaEvolutionEnabled is true?

Reply:
Sure will do...